docs/types/reference-types.rst

```diff
@@ -153,7 +153,8 @@ third-party string libraries. You can also compare two strings by their keccak25
 concatenate two strings using ``bytes.concat(bytes(s1), bytes(s2))``.
 
 You should use ``bytes`` over ``bytes1[]`` because it is cheaper,
-since ``bytes1[]`` adds 31 padding bytes between the elements. As a general rule,
+since using ``bytes1[]`` in ``memory`` adds 31 padding bytes between the elements. Note that in ``storage`` the
+padding is absent due to tight packing and hence both ``bytes1[]`` and ``bytes`` occupy the same space in ``storage``. As a general rule,
```

Comment:
>  hence both ``bytes1[]`` and ``bytes`` occupy the same space in ``storage``

With some caveats, `bytes` would get stored in place if the length is less than 31.

Reply:
Maybe you can remove the "...hence both" line, or add the additional information about in place encoding.

Reply:
I'm not sure I follow. Do you refer to fact that if bytes' size is at-most 31 bytes, it would be stored in place with the array(length * 2) slot?

Reply:
Yes, see https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html?highlight=stringlibytes-and-string#bytes-and-string.

Reply:
Yes, I now follow and will make changes.

Reply:
I have made the changes and added more info about the storage encoding.